src/domManipulation.ts

```diff
@@ -68,16 +79,41 @@ function temperatureColourConverter(temp) {
   }
 }
 
-export function displayForecastLoader(location) {
+export function displayForecastLoader(location: string) {
   if (location === "custom") {
-    document.getElementById("popup-loading-container").style.visibility = "visible";
-    document.getElementById("popup-loading-container").style.display = "flex";
-    document.getElementById("popup-forecast-container").style.visibility = "hidden";
-    document.getElementById("popup-forecast-container").style.display = "none";
+
+    const popupLoader = document.getElementById("popup-loading-container");
+
```

Comment:
As stated before you could use a guard clause to throw an error if these elements are null. This follows for all other `if (____ !== null)`

Reply:
Just be aware, if you throw, you must also catch somewhere